ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py

```diff
@@ -115,6 +115,21 @@ def test_numpy_ndarray_property_T(
         ground_truth_backend="numpy",
     )
 
+def test_numpy_ndarray_property_flat(
+    dtype_x,
+    item,
+):
+    dtype, data, shape = dtype_x
+    x = ndarray(shape, dtype[0])
+    x.ivy_array = data[0]
+    ivy.utils.assertions.check_equal(x.flat[item],data[0].flat[item],asarray=False)
+
+@given(
+    dtype_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid", prune_function=False),
+        ret_shape=True,
+    ),
```

Comment:
Please use helpers to generate the argument `item` in the decorator `@given` as well, since you are also using `item` to call the property `.flat`. Thank you 😄

Reply:
@given(
    dtype_x=helpers.dtype_and_values(
        available_dtypes=helpers.get_dtypes("valid", prune_function=False),
        ret_shape=True,
    ),
    item=helpers.item(
        available_items = helpers.get_items("valid", prune_function=False), 
        ret_shape=True,
    ),
)    
Can you see if this code is correct?

Reply:
@given(
    dtype_x=helpers.dtype_and_values(
        available_dtypes=helpers.get_dtypes("valid", prune_function=False),
        ret_shape=True,
    ),
    item=helpers.item(
        available_items = helpers.ints(1,dtype_x[3]),
        ret_shape=True,
    ),
)    
What about it now?

Reply:
![image](https://github.com/unifyai/ivy/assets/70466868/18467fd4-d538-4f4a-950d-babfcb2481c3)

Reply:
This is the function that i used....it looks fine to me but wanted a confirmation from you...sorry to disturb you again and again

Reply:
You can check this section on [Setting up Pytest](https://unify.ai/docs/ivy/overview/contributing/setting_up.html#setting-up-testing-in-pycharm)

Reply:
![image](https://github.com/unifyai/ivy/assets/70466868/9f2943e8-71f1-421e-9dda-1f4ec30bc253)

Reply:
What about this?

Reply:
> You can check this section on [Setting up Pytest](https://unify.ai/docs/ivy/overview/contributing/setting_up.html#setting-up-testing-in-pycharm)

I am not able to implement this. I am using VS Code and after setting up the environment I am confused how should i run local tests

Reply:
@Faiq2001 to run frontend tests, try these command
```bash
# run all the tests in a file
pytest ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py
# run a specific test, which is the function you're implementing
pytest ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py::test_numpy_ndarray_property_flat
```
The general format is
```bash
pytest test_file_path::test_fn_name
```

Reply:
@xoiga123 why am i getting errors while running these codes.
![image](https://github.com/unifyai/ivy/assets/70466868/4a8df333-ad48-4b2d-85f9-f34d8e64656d)

Reply:
I resolved some of the errors which were coming when importing 'config' from setuptools in __init__.py. But the errors just keep on increasing

Reply:
@Faiq2001 Sorry I can't merge this unless it's all good to go :mask:. It's fine if you have implemented the test successfully without setting up locally and verifying it, but currently the tests are not runnable on my end. It takes a little time to debug around and set things up but they will hugely save you time in the long run. You can check the [vscode setup channel](https://discord.com/channels/799879767196958751/1029037632245739610) if you need help :+1:, sorry that I won't be able to help much with vscode because I use pycharm :sob: But I think the docs are great for this.

Reply:
> Also dont you think the following helper function is correct? Item takes integers randomly between 1 to size(which is the total number of elements).
> 
> > ![image](https://user-images.githubusercontent.com/70466868/245153025-9f2943e8-71f1-421e-9dda-1f4ec30bc253.png)

Please try to generate arguments in the `@given` decorator only. You can watch this fantastic [Ivy Tests youtube video](https://www.youtube.com/watch?v=2AwWuHIe2h8&list=PLwNuX3xB_tv9NPbAVHs0dLe3GQkR51vG9&index=16) to get a hang of things and how to write a test properly :muscle:

Reply:
![image](https://github.com/unifyai/ivy/assets/70466868/8006a88b-5ac9-4d57-b5ff-4a67c0882ab2)
Code
![image](https://github.com/unifyai/ivy/assets/70466868/7727d7b6-c914-4b4a-a29c-9af87f0bd28a)
Tester
Changed the tester function and the code function as well, but still not able to run the tests locally. Can you guide me as to where i am making a mistake

Reply:
@xoiga123 I think i somehow unknowingly requested someone. It doesnt matter noh?